web/src/app/util/DestinationChip.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import { Avatar, Chip, CircularProgress } from '@mui/material'
+
+import {
+  BrokenImage,
+  RotateRight as RotationIcon,
+  Today as ScheduleIcon,
+  Webhook as WebhookIcon,
+} from '@mui/icons-material'
+import { DestinationDisplayInfo } from '../../schema'
+
+export type DestinationChipProps = DestinationDisplayInfo & {
+  error?: string
+
+  // If onDelete is provided, a delete icon will be shown.
+  onDelete?: () => void
+}
+
+const builtInIcons: { [key: string]: React.ReactNode } = {
+  'builtin://rotation': <RotationIcon />,
+  'builtin://schedule': <ScheduleIcon />,
+  'builtin://webhook': <WebhookIcon />,
+}
+
+/**
+ * DestinationChip is used to display a selected destination value.
+ *
+ * You should almost never use this component directly. Instead, use
+ * DestinationInputChip, which will select the correct values based on the
+ * provided DestinationInput value.
+ */
+export default function DestinationChip(
+  props: DestinationChipProps,
+): React.ReactNode {
+  if (props.error) {
+    return (
+      <Chip
+        avatar={
+          <Avatar>
+            <BrokenImage />
+          </Avatar>
+        }
+        label={'ERROR: ' + props.error}
+        onDelete={
+          props.onDelete
+            ? (e) => {
+                e.stopPropagation()
+                e.preventDefault()
+                props.onDelete?.()
+              }
+            : undefined
+        }
+      />
+    )
+  }
+  if (!props.text) {
+    return (
+      <Chip
+        avatar={
+          <Avatar>
+            <CircularProgress size='1em' />
+          </Avatar>
+        }
+        label='loading...'
+        onDelete={
+          props.onDelete
+            ? (e) => {
+                e.stopPropagation()
+                e.preventDefault()
+                props.onDelete?.()
+              }
+            : undefined
```

Comment:
I've been told not to use undefined explicitly, so I will do the same here

```suggestion
          props.onDelete
            ? (e) => {
                e.stopPropagation()
                e.preventDefault()
                props.onDelete?.()
              }
            : null
```

Reply:
I'm afraid this is not an option because the mui chips component expects the `onDelete` prop to be a function or undefined.
<img width="656" alt="Screenshot 2024-01-10 at 10 15 51 AM" src="https://github.com/target/goalert/assets/52386267/bd110cc4-4f17-459e-816f-ac49c49d898d">

Reply:
I agree! Just going off of precedent.